ROADMAP.md

```diff
@@ -0,0 +1,233 @@
+Roadmap towards full integration of linear types in GHC
+=======================================================
+
+This document outlines the roadmap for integrating linear types in
+GHC. It is laid out in stages which we intend to be successive merge
+points into the GHC tree. The _miscellaneous_ section contains issues
+which are parallel to the stages.
+
+The roadmap contains concrete implementation actions as well as more
+open discussion that needs to be better understood. The _remarks_
+section does not contain action, rather items that will eventually be
+documented.
+
+Stage 1: mvp
+------------
+
+The first iteration implements implements linear types purely in the
+type inference (no modification to core or its linter), and has
+monomorphic multiplicities
+
+- [x] Keep track of linearity in equation binders
+- [x] Constructors are linear by default
+- [x] Wildcard patterns rejected in the linear case
+- [ ] Constructors support unrestricted argument (`Unrestricted`)
+- [ ] Full justification for memory-safety in presence of exceptions
+- [ ] Syntax
+    - [ ] Syntax for multiplicity-annotated arrows. Options:
+        - Mimic Latex and write `a ->_l b` (currently used in this
+          roadmap, but this is no suggestion that this is the one to
+          be favoured)
+        - Edsko de Vries [proposes][edsko-blog] `a ->:l b` (with `l` a
+          multiplicity)
+        - …
+    - [ ] Do we need an ASCII short-hand for linear arrow `a ->_1 b`
+         (Unicode has `⊸` which is the standard notation for the
+         linear arrow)? Many proposals seen so far:
+        - Currently illegal: `-o`
+        - Currently legal: `-.`, `-*`, `-:`
```

Comment:
Even though they are 'legal' they have the wrong precedence and thus confuse readers